airbyte-integrations/connectors/source-freshdesk/source_freshdesk/api.py

```diff
@@ -138,15 +138,58 @@ def _api_get(self, url: str, params: Mapping = None):
     def list(self, fields: Sequence[str] = None) -> Iterator[dict]:
         """Iterate over entities"""
 
+    @property
+    def name(self):
+        """Name of the stream"""
+        stream_name = self.__class__.__name__
+        if stream_name.endswith("API"):
+            stream_name = stream_name[:-3]
+        return stream_name
+
     def read(self, getter: Callable, params: Mapping[str, Any] = None) -> Iterator:
         """Read using getter"""
         params = params or {}
-        for page in range(1, self.maximum_page):
-            batch = list(getter(params={**params, "per_page": self.result_return_limit, "page": page}))
-            yield from batch
 
-            if len(batch) < self.result_return_limit:
-                break
+        stream = self.name
+        # This block extends TicketsAPI Stream to overcome '300 page' server error.
+
+        if stream == "Tickets":
```

Comment:
nit: can we leave a comment here similar to this PR description explaining why we are taking this approach?

Reply:
We discussed the solution here in the issue: https://github.com/airbytehq/airbyte/issues/3858
The approach for this was taken into consideration as effective and easiest.

If you have any other ideas, please place them here or in the issue. Thanks for your comments.

Reply:
the approach is fine. this comment is more about leaving context within the code through comments so a future reader doesn't have to go and look for github issues to understand why/what we are doing here.

Reply:
Oh, I see, good point! May I take yours?)

Reply:
just saw this; yes!